erc20/vyper/vyper-erc20-spec.ini

```diff
@@ -131,60 +144,122 @@ requires:
+    ensures select(M2, #hashedLocation({COMPILER}, {_BALANCES}, FROM_ID)) ==Int BAL_FROM -Int VALUE
+    andBool select(M2, #hashedLocation({COMPILER}, {_BALANCES}, TO_ID)) ==Int BAL_TO +Int VALUE
+    andBool select(M2, #hashedLocation({COMPILER}, {_ALLOWANCES}, FROM_ID CALLER_ID)) ==Int ALLOW -Int VALUE
+    andBool M1 ==IMap M2 except
+      (SetItem(#hashedLocation({COMPILER}, {_BALANCES}, FROM_ID))
+       SetItem(#hashedLocation({COMPILER}, {_BALANCES}, TO_ID)) 
+       SetItem(#hashedLocation({COMPILER}, {_ALLOWANCES}, FROM_ID CALLER_ID)) .Set)
+
+[transferFrom-success-1-c]
++requires:
+    andBool #hashedLocation({COMPILER}, {_ALLOWANCES}, FROM_ID CALLER_ID) =/=Int #hashedLocation({COMPILER}, {_BALANCES}, FROM_ID)
+    andBool #hashedLocation({COMPILER}, {_ALLOWANCES}, FROM_ID CALLER_ID) ==Int #hashedLocation({COMPILER}, {_BALANCES}, TO_ID)
+ensures:
+    ensures select(M2, #hashedLocation({COMPILER}, {_BALANCES}, FROM_ID)) ==Int BAL_FROM -Int VALUE
+    andBool select(M2, #hashedLocation({COMPILER}, {_BALANCES}, TO_ID)) ==Int BAL_TO -Int VALUE
+    andBool select(M2, #hashedLocation({COMPILER}, {_ALLOWANCES}, FROM_ID CALLER_ID)) ==Int ALLOW -Int VALUE
+    andBool M1 ==IMap M2 except
+      (SetItem(#hashedLocation({COMPILER}, {_BALANCES}, FROM_ID))
+       SetItem(#hashedLocation({COMPILER}, {_BALANCES}, TO_ID)) 
+       SetItem(#hashedLocation({COMPILER}, {_ALLOWANCES}, FROM_ID CALLER_ID)) .Set)
+
 [transferFrom-success-2]
-storage:
-    #hashedLocation({COMPILER}, {_BALANCES},   FROM_ID)           |-> BAL_FROM
-    #hashedLocation({COMPILER}, {_ALLOWANCES}, FROM_ID CALLER_ID) |-> (ALLOW => ALLOW -Int VALUE)
-    _:Map
+storage: M1:IMap => M2:IMap
+origStorage: M1
 +requires:
+    andBool select(M1, #hashedLocation({COMPILER}, {_BALANCES}, FROM_ID)) ==Int BAL_FROM
+    andBool select(M1, #hashedLocation({COMPILER}, {_ALLOWANCES}, FROM_ID CALLER_ID)) ==Int ALLOW
     andBool FROM_ID ==Int TO_ID
     andBool VALUE <=Int BAL_FROM
     andBool VALUE <=Int ALLOW
+ensures:
+    ensures select(M2, #hashedLocation({COMPILER}, {_ALLOWANCES}, FROM_ID CALLER_ID)) ==Int ALLOW -Int VALUE
+    andBool M1 ==IMap M2 except
+      (SetItem(#hashedLocation({COMPILER}, {_ALLOWANCES}, FROM_ID CALLER_ID)) .Set)
 
 [transferFrom-failure]
-k: #execute => #halt
 statusCode: _ => EVMC_REVERT
 output: _ => _
 log: _
 
 [transferFrom-failure-1]
-storage:
-    #hashedLocation({COMPILER}, {_BALANCES},   FROM_ID)           |-> (BAL_FROM => _)  // BAL_FROM
-    #hashedLocation({COMPILER}, {_BALANCES},   TO_ID)             |-> (BAL_TO   => _)  // BAL_TO
-    #hashedLocation({COMPILER}, {_ALLOWANCES}, FROM_ID CALLER_ID) |-> ALLOW
-    _:Map
+storage: M1:IMap => M2:IMap
+origStorage: M1
++requires:
+    andBool select(M1, #hashedLocation({COMPILER}, {_BALANCES}, FROM_ID)) ==Int BAL_FROM
+    andBool select(M1, #hashedLocation({COMPILER}, {_BALANCES}, TO_ID)) ==Int BAL_TO
+    andBool select(M1, #hashedLocation({COMPILER}, {_ALLOWANCES}, FROM_ID CALLER_ID)) ==Int ALLOW
```

Comment:
nitpick: follow the vertical alignment.  see the original spec for the storage.

Reply:
I fixed this.

Reply:
no you didn't.  what did you fix?